src/charm.py

```diff
@@ -161,6 +162,22 @@ def _add_osd_action(self, event: ActionEvent):
 
         event.set_results({"status": "success"})
 
+    def _set_pool_size_action(self, event: ActionEvent):
+        """Set the size for one or more pools."""
+        pools = event.params.get("pools")
+        size = event.params.get("size")
+
+        cmd = ["sudo", "microceph", "pool", "set-rf", "--size", str(size), pools]
+        try:
+            logging.debug("Setting the new pool(s) size")
+            subprocess.run(cmd, check=True, timeout=180)
```

Comment:
The `_run_cmd` method in `microceph.py` has inbuilt logging for success and failure cases and ideally I'd prefer all CLI command calls to be organised in that file as that makes it simpler for mocking and keeps logically bound methods together.

Reply:
Good idea. I've refactored things a bit.